### PR TITLE
i18n

### PR DESCRIPTION
**Pull request by austin-schick**
Adds Spanish and German support to Desktop CMU Graphics.

Working on this PR I realized that we should
- Improve Desktop CMU Graphics typechecking (sometimes in the JS version we get away with typechecking late because `int / str` doesn't cause a runtime error)
- Improve the behavior of the console thread when errors are encountered (in the console or the main thread)

But I'll tackle those in a separate PR.

This PR also excludes translation of errors from Python itself. We'll need to collect a list of those somehow, and they vary per interpreter. I'm not sure what the best path forward is there.

### CODE REVIEW

**Review by schmave — APPROVED** [review]
Looks reasonable! Can you document the steps needed to add/modify/remove t() strings?

**Comment by schmave** [comment]
I ran the tests locally and they passed. Might be worth converting at least one of the tests to German, but that's your call.

File: tests/test_image_gen.py [hunk 1]
```diff
@@ -92,16 +92,23 @@ def run_test(driver, test_name, all_source_code):
...
+            correct_path = correct_path_fmt % (test_name[:-3], i)
+
         output_path = 'image_gen/%s/output_%d.png' % (test_name, i)
 
         source_code = ''
         source_code += 'import sys'
         source_code += '\nimport os'
-        if sys.platform == 'darwin':
-            source_code += '\nos.environ["SDL_VIDEODRIVER"] = "dummy"'
+        # if sys.platform == 'darwin':
```
[the remark this answers is not in the document]
Comment (austin-schick) [reply tests/test_image_gen.py:105]: Oops! That code shouldn't be commented

**Comment by austin-schick** [comment]
The auto-update tests are being flaky, and after running the auto-updater a few times on my computer I think we may need a different approach. I've just left the tests disabled for now

**Pull request merged**